### ISSUE TITLE
resolvedTheme can be `system`

### ISSUE THREAD

**Issue by pizzafox** [pacocoursey/next-themes#33]
I have noticed in testing that `useTheme().resolvedTheme` can be `system`. I think it's only happening in the first render, but I haven't tried making a minimal reproduction.

I am using next-themes@npm:0.0.13-beta.2 [`9e7cc`] (via npm:beta [`9e7cc`]) and not passing any options to `<ThemeProvider>`.

Is this expected behavior?

**Comment by thien-do** [comment]
Yeah this is really weird.. I also see this behavior, but by reading the code it should not happen https://github.com/pacocoursey/next-themes/blob/master/index.tsx !

**Comment by thien-do** [comment]
I also see it on 0.0.13, and I can reproduce it by setTheme("system") again after the initial set

**Comment by pizzafox** [comment]
I think it also happened on the latest stable version (which is what prompted me to update to beta). Maybe it has to do with the default theme being `system` + media query not working in SSR???

**Comment by thien-do** [comment]
Yeah the latest version doesn't change the resolving part so I think it likely is.

Actually I cannot reproduce your exact issue (resolvedTheme returns "system") but I do see some related issue I noted at https://github.com/pacocoursey/next-themes/issues/34 it's when the "system" is being used for "data-theme"

**Comment by pacocoursey** [comment]
Thank you both for reporting this! I just published `v0.0.13-beta.3` which should fix both #33 and #34 – please test and let me know if there is still an issue.

**Comment by pizzafox** [comment]
Updating to the latest beta seems to have fixed the issue! I'm curious, what was the source of this bug?

**Comment by pacocoursey** [comment]
Sweet, thanks! I wasn't able to reproduce the issue on the first render, but I was able to reproduce the issue when doing `setTheme('system')` twice. 

@thien-do explained it in #34 
> since the theme value (in state) is not changed (still "system"), the "handleMediaQuery" is not triggered

I updated the logic in this commit: https://github.com/pacocoursey/next-themes/pull/30/commits/8ab93d96b771cd4a523688d2cc6abce2386dada3

Basically when setting theme to system, it wasn't applying the resolved theme – that only happened by accident! This adds the check and actually makes it faster (previously there were two calls: one to update the DOM to system, then another to update the DOM to dark/light)
